### ISSUE TITLE
Multiple Assets can confuse vssetup

### ISSUE THREAD

**Issue by BenjamenMeyer** [vegastrike/Vega-Strike-Engine-Source#111]
Filing as part of https://github.com/vegastrike/Vega-Strike-Engine-Source/issues/98#issuecomment-626981181 on behalf of @stephengtuggy 

> There are multiple copies of the asset set now, and I have 3 of them checked out on my local machine. Vega Strike (specifically vssetup) is getting mixed up about which copy to use. The algorithm for determining the asset folder location, is not that smart.

**Comment by BenjamenMeyer** [comment]
The VS configuration utility should be able to handle multiple asset packages being on the same system.

**Comment by Loki1950** [comment]
Never intended for multiple asset sets so indeed the algorithm is primitive

**Comment by royfalk** [comment]
VSSetup is obsolete and AFAIK, Assets-Production is no longer a sub-module. 
@stephengtuggy  @BenjamenMeyer Can we close?

**Comment by BenjamenMeyer** [comment]
@royfalk does our new configuration tool resolve the issue?

**Comment by royfalk** [comment]
I am not 100% clear on what the issue was as I build from the source and don't install. Maybe @evertvorster or @stephengtuggy can answer that.

**Comment by BenjamenMeyer** [comment]
IIRC it was an issue with dealing with multiple games - PWCU vs VS:UtCS, etc.
We really should split config into: (a) core game engine and (b) game asset config; and then allow (1) the game asset config to be separately managed and (2) perhaps allow multiple game engine config profiles that the game assets could select between in case users want to configure the core engine differently for each game, but then make it easy to share the config between the games.

If it's not 100% clear on the issue we can certainly close this and start some new issues around how to make the user experience nicer when one has multiple games installed.

**Comment by stephengtuggy** [comment]
This has been fixed for a long time, from my perspective. You would simply pass the assets directory to vegasettings on the command line, and it worked great. I think we can close this.

**Comment by royfalk** [comment]
> You would simply pass the assets directory to vegasettings on the command line

Sorry for reopening.

vegasettings has been deprecated. If we transition to making settings part of the main game, we can't rely on it to run the game...

**Comment by BenjamenMeyer** [comment]
> > You would simply pass the assets directory to vegasettings on the command line
> 
> Sorry for reopening.
> 
> vegasettings has been deprecated. If we transition to making settings part of the main game, we can't rely on it to run the game...

Correct we will need to ensure there is a base level of operational capability for the user to be able to get to the configuration capability. That would probably mean starting by default in a windows mode of a default size/resolution.

**Comment by royfalk** [comment]
Default resolution isn't the problem. If I have two VS games installed and I run vegastrike-engine in order to play the game, which will it choose?
We should have a mechanism to publish games (Assets) to the engine or some other way of presenting choice to the users.
As I've said, I'm not sure how users "consume" this game.

**Comment by BenjamenMeyer** [comment]
> Default resolution isn't the problem. If I have two VS games installed and I run vegastrike-engine in order to play the game, which will it choose?

Each has their own configuration directory. For example, VS:UtCS uses `~/.vegastrike` while PWCU uses `~/.pu`; Privateer Gold used `~/.privgold100`. This gets configured via the assets in some form. (The `Version.txt` file in the Assets sets this.)

> We should have a mechanism to publish games (Assets) to the engine or some other way of presenting choice to the users. As I've said, I'm not sure how users "consume" this game.

Yes, I'd like to have a launcher to support multiple assets for the user to select between so we have an easier integration. The launcher config could take care of the baseline configuration.

**Comment by stephengtuggy** [comment]
> Default resolution isn't the problem. If I have two VS games installed and I run vegastrike-engine in order to play the game, which will it choose? We should have a mechanism to publish games (Assets) to the engine or some other way of presenting choice to the users. As I've said, I'm not sure how users "consume" this game.

You pass the game directory to vegastrike-engine on the command line using the `-d` or `--target` parameter. Then a GUI launcher could take care of those users that don't want to bother with the command line. And/or we have each set of game assets create its own application shortcut(s) that pass their respective assets directory to vegastrike-engine on the command line within the shortcut properties. Assets-Production (VS:UtCS) v0.9.x already does this.

**Comment by BenjamenMeyer** [comment]
I don't think this is an issue any more; as noted the vegasettings application that was originally provided has been deprecated going forward (0.10.x and later); and this is already working properly in 0.9.x.

We can reopen if folks are still having an issue with 0.9.x. For 0.10.x and later we can file a new issue for what is needed.
